gem_city.rb

```diff
@@ -0,0 +1,68 @@
+class TypeError < StandardError; end
+
+# start GemCity class
+class GemCity
+  attr_reader :thieves, :officers, :population, :civilians
+  # This class represents the town of GemCity
+  # This is a town riddled with crime but we can find out how happy the town is
+
+  def initialize
+    @thieves = 5
+    @officers = 1
+    @population = 50
+    @civilians = @population - @thieves - @officers
+  end
+
+  def thieves=(number)
+    fail TypeError, 'Input must be an integer' unless number.is_a? Integer
+    @thieves = number
+  end
+
+  def officers=(number)
+    fail TypeError, 'Input must be an integer' unless number.is_a? Integer
+    @officers = number
+  end
+
+  def civilians=(number)
+    fail TypeError, 'Input must be an integer' unless number.is_a? Integer
+    @civilians = number
+  end
+
+  def population=(number)
+    fail TypeError, 'Input must be an integer' unless number.is_a? Integer
+    @population = number
+  end
+
+  def demographics
+    @demographics = { 'thieves' => rounded_percentage(thieves),
+                      'officers' => rounded_percentage(officers),
+                      'civilians' => rounded_percentage(civilians) }
+  end
+
+  def happiness_of_town
+    # happiness is random... people don't know what they want!
+    happiness_values = []
+    happiness = 0
+    counter = 0
+    while counter < population
```

Comment:
You can simplify this block down to one line by using a Range and ``#each`` iterator